Userland/Libraries/LibCore/ArgsParser.cpp

```diff
@@ -111,7 +111,7 @@ bool ArgsParser::parse(Span<StringView> arguments, FailureBehavior failure_behav
         VERIFY(found_option);
 
```

Comment:
...both of which make me question what precisely you mean by the third failure state being a system problem.
Ultimately all failures at this point are user problems, "you have too many arguments", "your input isn't valid utf-8", "your input isn't a number", "I expected 'foo' or 'bar', not 'baz'", etc.

Reply:
Hmm, the system failure I am thinking of here is an allocation failure, which I don't really think is the users fault and I'm not quite sure how to gracefully handle that _other than_ exiting the program and saying that we are out of memory.